lib/posix/fs.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <errno.h>
+#include <posix/unistd.h>
+
+
+struct p_fs_file {
+	struct fs_file_t file;
+	bool used;
+};
+
+struct p_fs_dir {
+	struct fs_dir_t	dir;
+	bool used;
+};
+
+static struct p_fs_dir dir_array[CONFIG_MAX_OPEN_FILES];
+K_MUTEX_DEFINE(dirp_mutex);
+
+static struct p_fs_file file_array[CONFIG_MAX_OPEN_FILES];
+K_MUTEX_DEFINE(filep_mutex);
+
+static struct fs_dirent fdirent;
+static struct dirent pdirent;
+
+/**
+ * @brief Open a file.
+ *
+ * See IEEE 1003.1
+ */
+int open(const char *name, int flags)
+{
+	int rc, fd;
+	struct fs_file_t *ptr;
+
+	ARG_UNUSED(flags);
+
+	k_mutex_lock(&filep_mutex, K_FOREVER);
+	/* Find free file descriptor */
+	for (fd = 0; fd < CONFIG_MAX_OPEN_FILES; fd++) {
+		if (file_array[fd].used == false) {
+			ptr = &file_array[fd].file;
+			memset(ptr, 0, sizeof(struct fs_file_t));
```

Comment:
Performance nit: put the memset and all initialization outside the lock.  The locking should be to find an open slot and reserve it.  Then you can touch the file record without synchronization because nothing else will use it.